runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3196,7 +3199,78 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 		extensions->gcThreadCountForced = false;
 	}
 
-	return result;
+	/* While ideally it is desired to change the heap geometry (parameters such as
+	 * extensions->maxMemory, virtual memory reserved, etc), it would be too complex
+	 * and slow. As an alternative, the geometry remains unchanged, but the
+	 * existing softmx functionality is utilized to limit the maximum size of heap.
+	 */
+
+	{
+		uintptr_t candidateSoftMx = 0;
+		if (0.0 <= extensions->maxRAMPercent) {
+			candidateSoftMx = extensions->maxRAMPercent * extensions->usablePhysicalMemory / 100.0;
+		} else {
+			/* False is used for computeDefaultMaxHeapForJava(), since in restore
+			 * scenario enableOriginalJDK8HeapSizeCompatibilityOption is always false.
+			 */
+			candidateSoftMx = extensions->computeDefaultMaxHeapForJava(false);
+		}
+		if (extensions->dynamicHeapAdjustment) {
+			/* clear up the existing softMx value because of the dynamicHeapAdjustment is set. */
+			if (extensions->memoryMax > candidateSoftMx) {
+				/* If the softmx value is smaller than Xms, it is set to Xms value. */
+				if (extensions->initialMemorySize > candidateSoftMx) {
+					candidateSoftMx = extensions->initialMemorySize;
+				} else {
+					/* If the candidate softmx value is within [Xms, Xmx],
+					 * the softmx is assigned to candidate softmx.
+					 */
+				}
+			} else {
+				candidateSoftMx = 0;
```

Comment:
you can put a comment like: adjusting to use the whole available heap, but instead of setting softmx to mx value, softmx functionality is disabled